node/src/taiko/mod.rs

```diff
@@ -727,11 +735,16 @@ fn calculate_max_bytes_per_tx_list(
     max_bytes_per_tx_list: u64,
     throttling_factor: u64,
     batches_ready_to_send: u64,
+    min_pending_tx_list_size_bytes: u64,
 ) -> u64 {
     let mut size = max_bytes_per_tx_list;
     for _ in 0..batches_ready_to_send {
         size = size.saturating_sub(size / throttling_factor);
     }
+    size = min(
+        max_bytes_per_tx_list,
+        max(size, min_pending_tx_list_size_bytes),
+    );
```

Comment:
Why not just:

```suggestion
    size = max(size, min_pending_tx_list_size_bytes);
```

?